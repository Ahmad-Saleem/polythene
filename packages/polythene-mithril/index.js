export * from "polythene-mithril-base";
export * from "polythene-mithril-button";
export * from "polythene-mithril-fab";
export * from "polythene-mithril-icon";
export * from "polythene-mithril-raised-button";
export * from "polythene-mithril-ripple";
export * from "polythene-mithril-shadow";
export * from "polythene-mithril-svg";
