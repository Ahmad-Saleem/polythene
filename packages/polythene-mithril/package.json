{
  "name": "polythene-mithril",
  "version": "1.0.0-rc.1",
  "description": "",
  "main": "dist/polythene-mithril.js",
  "module": "dist/polythene-mithril.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "npm run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-mithril-base": "1.0.0-rc.1",
    "polythene-mithril-button": "1.0.0-rc.1",
    "polythene-mithril-card": "1.0.0-rc.1",
    "polythene-mithril-checkbox": "1.0.0-rc.1",
    "polythene-mithril-dialog": "1.0.0-rc.1",
    "polythene-mithril-dialog-pane": "1.0.0-rc.1",
    "polythene-mithril-fab": "1.0.0-rc.1",
    "polythene-mithril-icon": "1.0.0-rc.1",
    "polythene-mithril-icon-button": "1.0.0-rc.1",
    "polythene-mithril-list": "1.0.0-rc.1",
    "polythene-mithril-list-tile": "1.0.0-rc.1",
    "polythene-mithril-menu": "1.0.0-rc.1",
    "polythene-mithril-radio-button": "1.0.0-rc.1",
    "polythene-mithril-raised-button": "1.0.0-rc.1",
    "polythene-mithril-ripple": "1.0.0-rc.1",
    "polythene-mithril-shadow": "1.0.0-rc.1",
    "polythene-mithril-svg": "1.0.0-rc.1",
    "polythene-mithril-switch": "1.0.0-rc.1",
    "polythene-mithril-tabs": "1.0.0-rc.1",
    "polythene-mithril-textfield": "1.0.0-rc.1",
    "polythene-mithril-toolbar": "1.0.0-rc.1"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://visiblearea.com)",
  "license": "MIT"
}

