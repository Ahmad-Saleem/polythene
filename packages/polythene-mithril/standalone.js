export * from "./index";
import { addLayoutStyles, addFastClick } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-material-design";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
