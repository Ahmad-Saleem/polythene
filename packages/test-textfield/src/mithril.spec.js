import { runSnapshots } from "../../../scripts/tests/mithril-snapshots";
import { renderer, TextField } from "polythene-mithril";
import specTests from "./spec-tests.js";
import mithrilTests from "./tests-mithril.js";

runSnapshots({
  tests: specTests({ TextField, renderer }).concat(mithrilTests),
  renderer
});
