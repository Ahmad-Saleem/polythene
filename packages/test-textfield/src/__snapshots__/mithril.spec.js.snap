exports[`test Custom validation (only use lowercase characters) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"text\">
    </div>
    <div class=\"pe-textfield__error-placeholder\"></div>
  </div>
</div>
"
`;

exports[`test Dark tone class + light theme class 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear pe-light-tone\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      Name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Dark tone class + light tone 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear pe-light-tone\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      Name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Give focus 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div>
    <div class=
    \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
    <div class=\"pe-textfield__input-area\">
        <label class=\"pe-textfield__label\">Your
        name</label><input class=\"pe-textfield__input\" type=\"text\">
      </div>
    </div><a class=\"pe-button pe-text-button pe-raised-button\"
    tabindex=\"0\">
    <div class=\"pe-button__content\">
      <div class=\"pe-shadow pe-shadow--animated\">
        <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
        <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
      </div>
      <div class=\"pe-ripple pe-ripple--constrained\"></div>
      <div class=\"pe-button__wash\"></div>
      <div class=\"pe-button__focus\"></div>
      <div class=\"pe-button__label\">
        Give focus
      </div>
    </div></a>
  </div>
</div>
"
`;

exports[`test No options 1`] = `
"<div class=
\"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
    <input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;

exports[`test Option: autofocus 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"text\" autofocus=\"\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: class 1`] = `
"<div class=
\"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear class-x\">
<div class=\"pe-textfield__input-area\">
    <input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;

exports[`test Option: className 1`] = `
"<div class=
\"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear className-x\">
<div class=\"pe-textfield__input-area\">
    <input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;

exports[`test Option: counter 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--floating-label pe-textfield--counter pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=
      \"pe-textfield__label\">Description</label><input class=
      \"pe-textfield__input\" type=\"text\">
    </div>
    <div class=\"pe-textfield__counter\">
      0 / 30
    </div>
  </div>
</div>
"
`;

exports[`test Option: counter, maxlength 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--floating-label pe-textfield--counter pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=
      \"pe-textfield__label\">Description</label><input class=
      \"pe-textfield__input\" type=\"text\" maxlength=\"30\">
    </div>
    <div class=\"pe-textfield__counter\">
      0 / 30
    </div>
    <div class=\"pe-textfield__error-placeholder\"></div>
  </div>
</div>
"
`;

exports[`test Option: defaultValue 1`] = `
"<div class=
\"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
<div class=\"pe-textfield__input-area\">
    <input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;

exports[`test Option: defaultValue 2`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: disabled (input) -- dark theme class 1`] = `
"<div style=\"padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--disabled pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" disabled type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: disabled (input) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--disabled pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" disabled type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: disabled (label) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--disabled pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      name</label><input class=\"pe-textfield__input\" disabled type=
      \"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: element 1`] = `
"<a class=
\"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
<div class=\"pe-textfield__input-area\">
  <input class=\"pe-textfield__input\" type=\"text\">
</div></a>
"
`;

exports[`test Option: floatingLabel -- dark theme class 1`] = `
"<div style=\"padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      email</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      address</label><input class=\"pe-textfield__input\" type=
      \"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: floatingLabel 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      email</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      address</label><input class=\"pe-textfield__input\" type=
      \"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: floatingLabel, dense 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--dense pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--dense pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      email</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--dense pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      address</label><input class=\"pe-textfield__input\" type=
      \"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: focusHelp 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      Name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
    <div class=\"pe-textfield__help pe-textfield__help-focus\">
      Enter the name as written on the credit card
    </div>
  </div>
</div>
"
`;

exports[`test Option: fullWidth 1`] = `
"<div style=\"background: rgb(255, 255, 255);\">
  <div class=
  \"pe-textfield pe-textfield--full-width pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--full-width pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      email</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--full-width pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      address</label><input class=\"pe-textfield__input\" type=
      \"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: help 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      Name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
    <div class=\"pe-textfield__help\">
      Enter the name as written on the credit card
    </div>
  </div>
</div>
"
`;

exports[`test Option: id 1`] = `
"<div id=\"id-x\" class=
\"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
    <input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;

exports[`test Option: label -- dark theme class 1`] = `
"<div style=\"padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      Name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: label 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your
      Name</label><input class=\"pe-textfield__input\" type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: maxlength 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"text\" maxlength=\"3\">
    </div>
    <div class=\"pe-textfield__error-placeholder\"></div>
  </div>
</div>
"
`;

exports[`test Option: min, max 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"number\" max=\"8\" min=
      \"3\" required=\"\">
    </div>
    <div class=\"pe-textfield__error-placeholder\"></div>
  </div>
</div>
"
`;

exports[`test Option: multiLine 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Label in multi-line
      input</label>
      <textarea class=\"pe-textfield__input\" rows=\"2\">
</textarea>
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Floating label in
      multi-line input</label>
      <textarea class=\"pe-textfield__input\" rows=\"2\">
</textarea>
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <textarea class=\"pe-textfield__input\" rows=\"4\">
</textarea>
    </div>
  </div>
</div>
"
`;

exports[`test Option: onChange 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div>
    <div class=
    \"pe-textfield pe-textfield--counter pe-textfield--hide-spinner pe-textfield--hide-clear\">
    <div class=\"pe-textfield__input-area\">
        <input class=\"pe-textfield__input\" type=\"text\">
      </div>
      <div class=\"pe-textfield__counter\">
        0 / 6
      </div>
    </div>
    <div style=\"margin: 10px 0px;\">
      <div>
        focus: undefined
      </div>
      <div>
        dirty: undefined
      </div>
      <div>
        invalid: undefined
      </div>
    </div><a class=\"pe-button pe-text-button pe-raised-button\"
    tabindex=\"0\">
    <div class=\"pe-button__content\">
      <div class=\"pe-shadow pe-shadow--animated\">
        <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
        <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
      </div>
      <div class=\"pe-ripple pe-ripple--constrained\"></div>
      <div class=\"pe-button__wash\"></div>
      <div class=\"pe-button__focus\"></div>
      <div class=\"pe-button__label\">
        Random
      </div>
    </div></a>
  </div>
</div>
"
`;

exports[`test Option: optionalIndicator (element) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your name<span class=
      \"pe-textfield__optional-indicator\"><span style=
      \"color: rgb(51, 51, 51);\">this field is
      optional</span></span></label><input class=
      \"pe-textfield__input\" type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: optionalIndicator (string) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your name<span class=
      \"pe-textfield__optional-indicator\">(optional)</span></label><input class=\"pe-textfield__input\"
      type=\"text\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: pattern [0-9]+, validateAtStart 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=
      \"pe-textfield__label\">Number</label><input class=\"pe-textfield__input\"
      type=\"text\" pattern=\"[0-9]+\">
    </div>
    <div class=\"pe-textfield__error-placeholder\"></div>
  </div>
</div>
"
`;

exports[`test Option: readonly -- dark theme class 1`] = `
"<div style=\"padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--readonly pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"text\" readonly>
    </div>
  </div>
</div>
"
`;

exports[`test Option: readonly 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--readonly pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"text\" readonly>
    </div>
  </div>
</div>
"
`;

exports[`test Option: required 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your name<span class=
      \"pe-textfield__required-indicator\">*</span></label><input class=\"pe-textfield__input\"
      type=\"text\" required=\"\">
    </div>
    <div class=\"pe-textfield__help\">
      Enter the name as written on the credit card
    </div>
  </div>
</div>
"
`;

exports[`test Option: required, floatingLabel 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your name<span class=
      \"pe-textfield__required-indicator\">*</span></label><input class=\"pe-textfield__input\"
      type=\"text\" required=\"\">
    </div>
    <div class=\"pe-textfield__help\">
      Enter the name as written on the credit card
    </div>
  </div>
</div>
"
`;

exports[`test Option: required, floatingLabel, empty requiredIndicator 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Floating
      label</label><input class=\"pe-textfield__input\" type=\"text\"
      required=\"\">
    </div>
    <div class=\"pe-textfield__help pe-textfield__help-focus\">
      This required field does not show a *
    </div>
  </div>
</div>
"
`;

exports[`test Option: required, floatingLabel, requiredIndicator (element) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your name<span class=
      \"pe-textfield__required-indicator\"><span style=
      \"color: rgb(51, 51, 51);\">this field is
      required</span></span></label><input class=
      \"pe-textfield__input\" type=\"text\" required=\"\">
    </div>
    <div class=\"pe-textfield__help\">
      Enter the name as written on the credit card
    </div>
  </div>
</div>
"
`;

exports[`test Option: required, floatingLabel, requiredIndicator (string) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--floating-label pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your name<span class=
      \"pe-textfield__required-indicator\">(required)</span></label><input class=\"pe-textfield__input\"
      type=\"text\" required=\"\">
    </div>
    <div class=\"pe-textfield__help\">
      Enter the name as written on the credit card
    </div>
  </div>
</div>
"
`;

exports[`test Option: required, hideValidation 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear pe-textfield--hide-validation\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Your name<span class=
      \"pe-textfield__required-indicator\">*</span></label><input class=\"pe-textfield__input\"
      type=\"text\" required=\"\">
    </div>
    <div class=\"pe-textfield__help\">
      Enter the name as written on the credit card
    </div>
  </div>
</div>
"
`;

exports[`test Option: style 1`] = `
"<div style=\"background: rgb(33, 150, 243);\" class=
\"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
    <label class=\"pe-textfield__label\">Your
    name</label><input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;

exports[`test Option: type (password, number, email) 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"password\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"number\">
    </div>
  </div>
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <input class=\"pe-textfield__input\" type=\"email\">
    </div>
  </div>
</div>
"
`;

exports[`test Option: type email, required -- dark theme class 1`] = `
"<div style=\"padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Email<span class=
      \"pe-textfield__required-indicator\">*</span></label><input class=\"pe-textfield__input\"
      type=\"email\" required=\"\">
    </div>
    <div class=\"pe-textfield__error-placeholder\"></div>
  </div>
</div>
"
`;

exports[`test Option: type email, required 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div class=
  \"pe-textfield pe-textfield--dirty pe-textfield--required pe-textfield--hide-spinner pe-textfield--hide-clear\">
  <div class=\"pe-textfield__input-area\">
      <label class=\"pe-textfield__label\">Email<span class=
      \"pe-textfield__required-indicator\">*</span></label><input class=\"pe-textfield__input\"
      type=\"email\" required=\"\">
    </div>
    <div class=\"pe-textfield__error-placeholder\"></div>
  </div>
</div>
"
`;

exports[`test Option: value 1`] = `
"<div class=
\"pe-textfield pe-textfield--dirty pe-textfield--hide-spinner pe-textfield--hide-clear\">
<div class=\"pe-textfield__input-area\">
    <input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;

exports[`test Set value 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div>
    <div class=
    \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
    <div class=\"pe-textfield__input-area\">
        <input class=\"pe-textfield__input\" type=\"text\">
      </div>
      <div class=\"pe-textfield__help\">
        Type text, or press ARROW RIGHT to insert a character
        programmaticaly
      </div>
    </div><a class=\"pe-button pe-text-button pe-raised-button\"
    tabindex=\"0\">
    <div class=\"pe-button__content\">
      <div class=\"pe-shadow pe-shadow--animated\">
        <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
        <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
      </div>
      <div class=\"pe-ripple pe-ripple--constrained\"></div>
      <div class=\"pe-button__wash\"></div>
      <div class=\"pe-button__focus\"></div>
      <div class=\"pe-button__label\">
        Clear
      </div>
    </div></a>
  </div>
</div>
"
`;

exports[`test Set value 2`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px 15px;\">
  <div>
    <div class=
    \"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear\">
    <div class=\"pe-textfield__input-area\">
        <input class=\"pe-textfield__input\" type=\"text\">
      </div>
      <div class=\"pe-textfield__help\">
        Type text, or press ARROW RIGHT to insert a character
        programmaticaly
      </div>
    </div><a class=\"pe-button pe-text-button pe-raised-button\"
    tabindex=\"0\">
    <div class=\"pe-button__content\">
      <div class=\"pe-shadow pe-shadow--animated\">
        <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
        <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
      </div>
      <div class=\"pe-ripple pe-ripple--constrained\"></div>
      <div class=\"pe-button__wash\"></div>
      <div class=\"pe-button__focus\"></div>
      <div class=\"pe-button__label\">
        Clear
      </div>
    </div></a>
  </div>
</div>
"
`;

exports[`test Themed 1`] = `
"<div class=
\"pe-textfield pe-textfield--hide-spinner pe-textfield--hide-clear tests-textfield-themed-textfield\">
<div class=\"pe-textfield__input-area\">
    <label class=\"pe-textfield__label\">Your
    name</label><input class=\"pe-textfield__input\" type=\"text\">
  </div>
</div>
"
`;
